### ISSUE TITLE
My device crashed when I was drawing

### ISSUE THREAD

**Issue by TatianaHde13** [opentoonz/opentoonz#5542]
### Description

Appeared this when  I opened that again
OpenToonz 1.7.1  (Build Mar 14 2023)

Report Date: 2024-07-20 17:35:54
Crash Reason: EXCEPTION_ACCESS_VIOLATION

Build ABI: x86_64-little_endian-llp64
Operating System: Windows 10 Version 2009
OS Kernel: 10.0.22621
CPU Threads: 2

GPU Vendor: ATI Technologies Inc.
GPU Model: AMD Radeon(TM) Graphics
GPU Version: 4.6.14761 Compatibility Profile Context 21.30.44.10 30.0.13044.10001

Crash File: C:\Users\tatia\AppData\Local\OpenToonz\OpenToonz\crash\Crash-20240720-173554.log
Mini Dump File: C:\Users\tatia\AppData\Local\OpenToonz\OpenToonz\crash\Crash-20240720-173554.dmp

Application Dir: C:/Program Files/OpenToonz
Stuff Dir: C:\OpenToonz stuff

Project Name: algo nc lore
Scene Name: algo
Project Path: C:\OpenToonz stuff\projects\algo nc lore\algo nc lore_otprj.xml
Scene Path: C:\OpenToonz stuff\projects\algo nc lore\scenes\algo.tnz

==== Modules ====
C:\Program Files\OpenToonz\OpenToonz.exe
C:\WINDOWS\SYSTEM32\ntdll.dll
C:\WINDOWS\System32\KERNEL32.DLL
C:\WINDOWS\System32\KERNELBASE.dll
C:\WINDOWS\System32\USER32.dll
C:\WINDOWS\System32\win32u.dll
C:\Program Files\OpenToonz\Qt5Xml.dll
C:\WINDOWS\System32\GDI32.dll
C:\WINDOWS\System32\gdi32full.dll
C:\WINDOWS\System32\msvcp_win.dll
C:\WINDOWS\System32\ucrtbase.dll
C:\Program Files\OpenToonz\Qt5PrintSupport.dll
C:\WINDOWS\System32\SHELL32.dll
C:\WINDOWS\System32\ole32.dll
C:\WINDOWS\System32\combase.dll
C:\WINDOWS\SYSTEM32\OPENGL32.dll
C:\WINDOWS\System32\RPCRT4.dll
C:\WINDOWS\System32\msvcrt.dll
C:\WINDOWS\System32\OLEAUT32.dll
C:\WINDOWS\System32\ADVAPI32.dll
C:\WINDOWS\System32\sechost.dll
C:\WINDOWS\System32\bcrypt.dll
C:\Program Files\OpenToonz\glut64.dll
C:\Program Files\OpenToonz\Qt5SerialPort.dll
C:\WINDOWS\System32\COMDLG32.dll
C:\WINDOWS\System32\shcore.dll
C:\WINDOWS\System32\SETUPAPI.dll
C:\WINDOWS\System32\SHLWAPI.dll
C:\Program Files\OpenToonz\EDSDK.dll
C:\Program Files\OpenToonz\colorfx.dll
C:\WINDOWS\System32\WS2_32.dll
C:\Program Files\OpenToonz\image.dll
C:\Program Files\OpenToonz\tnztools.dll
C:\Program Files\OpenToonz\tnzstdfx.dll
C:\Program Files\OpenToonz\tfarm.dll
C:\Program Files\OpenToonz\Qt5Gui.dll
C:\Program Files\OpenToonz\Qt5Core.dll
C:\Program Files\OpenToonz\Qt5Widgets.dll
C:\Program Files\OpenToonz\sound.dll
C:\Program Files\OpenToonz\toonzqt.dll
C:\Program Files\OpenToonz\opencv_world451.dll
C:\Program Files\OpenToonz\toonzlib.dll
C:\Program Files\OpenToonz\Qt5Script.dll
C:\Program Files\OpenToonz\tnzext.dll
C:\Program Files\OpenToonz\tnzbase.dll
C:\Program Files\OpenToonz\Qt5OpenGL.dll
C:\Program Files\OpenToonz\tnzcore.dll
C:\Program Files\OpenToonz\Qt5Multimedia.dll
C:\Program Files\OpenToonz\Qt5Network.dll
C:\WINDOWS\SYSTEM32\MPR.dll
C:\WINDOWS\System32\CRYPT32.dll
C:\WINDOWS\SYSTEM32\GLU32.dll
C:\WINDOWS\SYSTEM32\dbghelp.dll
C:\WINDOWS\SYSTEM32\MFPlat.DLL
C:\Program Files\OpenToonz\MSVCP140.dll
C:\Program Files\OpenToonz\VCRUNTIME140.dll
C:\WINDOWS\SYSTEM32\MF.dll
C:\Program Files\OpenToonz\VCRUNTIME140_1.dll
C:\WINDOWS\SYSTEM32\WINSPOOL.DRV
C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_5.82.22621.3527_none_b43b7f4b638cc64f\COMCTL32.dll
C:\WINDOWS\SYSTEM32\VERSION.dll
C:\WINDOWS\SYSTEM32\MSVFW32.dll
C:\WINDOWS\SYSTEM32\AVIFIL32.dll
C:\Program Files\OpenToonz\libmypaint-1-4-0.dll
C:\Program Files\OpenToonz\glew32.dll
C:\WINDOWS\SYSTEM32\dxgi.dll
C:\WINDOWS\SYSTEM32\d3d11.dll
C:\WINDOWS\SYSTEM32\USERENV.dll
C:\WINDOWS\SYSTEM32\NETAPI32.dll
C:\WINDOWS\SYSTEM32\WINMM.dll
C:\WINDOWS\SYSTEM32\dwmapi.dll
C:\WINDOWS\SYSTEM32\UxTheme.dll
C:\Program Files\OpenToonz\MSVCP140_1.dll
C:\Program Files\OpenToonz\Qt5Svg.dll
C:\Program Files\OpenToonz\CONCRT140.dll
C:\WINDOWS\SYSTEM32\DNSAPI.dll
C:\WINDOWS\SYSTEM32\IPHLPAPI.DLL
C:\WINDOWS\SYSTEM32\cfgmgr32.dll
C:\WINDOWS\SYSTEM32\MSACM32.dll
C:\Program Files\OpenToonz\libintl-8.dll
C:\Program Files\OpenToonz\libjson-c-2.dll
C:\WINDOWS\SYSTEM32\dxcore.dll
C:\WINDOWS\SYSTEM32\dbgcore.DLL
C:\Program Files\OpenToonz\libiconv-2.dll
C:\WINDOWS\SYSTEM32\NETUTILS.DLL
C:\WINDOWS\SYSTEM32\MFCORE.DLL
C:\WINDOWS\SYSTEM32\SRVCLI.DLL
C:\WINDOWS\SYSTEM32\CRYPTBASE.DLL
C:\WINDOWS\System32\IMM32.DLL
C:\WINDOWS\System32\NSI.dll
C:\WINDOWS\SYSTEM32\kernel.appcore.dll
C:\WINDOWS\System32\bcryptPrimitives.dll
C:\WINDOWS\SYSTEM32\RTWorkQ.DLL
C:\Program Files\OpenToonz\platforms\qwindows.dll
C:\WINDOWS\SYSTEM32\WTSAPI32.dll
C:\WINDOWS\SYSTEM32\windows.storage.dll
C:\WINDOWS\SYSTEM32\wintypes.dll
C:\WINDOWS\SYSTEM32\profapi.dll
C:\WINDOWS\SYSTEM32\powrprof.dll
C:\WINDOWS\SYSTEM32\UMPDC.dll
C:\Program Files\OpenToonz\styles\qwindowsvistastyle.dll
C:\Program Files\OpenToonz\iconengines\qsvgicon.dll
C:\WINDOWS\system32\d3d9.dll
C:\WINDOWS\SYSTEM32\directxdatabasehelper.dll
C:\WINDOWS\System32\MSCTF.dll
C:\WINDOWS\SYSTEM32\textinputframework.dll
C:\WINDOWS\SYSTEM32\CoreMessaging.dll
C:\WINDOWS\SYSTEM32\CoreUIComponents.dll
C:\WINDOWS\System32\clbcatq.dll
C:\Windows\System32\AppXDeploymentClient.dll
C:\WINDOWS\System32\DriverStore\FileRepository\u0377487.inf_amd64_99415ae859206efa\B377170\atig6pxx.dll
C:\WINDOWS\System32\DriverStore\FileRepository\u0377487.inf_amd64_99415ae859206efa\B377170\atio6axx.dll
C:\WINDOWS\SYSTEM32\dxva2.dll
C:\WINDOWS\System32\DriverStore\FileRepository\u0377487.inf_amd64_99415ae859206efa\B377170\atig6txx.dll
C:\WINDOWS\SYSTEM32\atiadlxx.dll
C:\WINDOWS\System32\PSAPI.DLL
C:\WINDOWS\SYSTEM32\PROPSYS.dll
C:\WINDOWS\SYSTEM32\pdh.dll
C:\WINDOWS\SYSTEM32\DEVOBJ.dll
C:\WINDOWS\System32\WINTRUST.dll
C:\WINDOWS\SYSTEM32\MSASN1.dll
C:\WINDOWS\system32\dataexchange.dll
C:\WINDOWS\system32\twinapi.appcore.dll
C:\WINDOWS\system32\dwrite.dll
C:\WINDOWS\SYSTEM32\WINMMBASE.dll
C:\Program Files\OpenToonz\imageformats\qgif.dll
C:\Program Files\OpenToonz\imageformats\qicns.dll
C:\Program Files\OpenToonz\imageformats\qico.dll
C:\Program Files\OpenToonz\imageformats\qjpeg.dll
C:\Program Files\OpenToonz\imageformats\qsvg.dll
C:\Program Files\OpenToonz\imageformats\qtga.dll
C:\Program Files\OpenToonz\imageformats\qtiff.dll
C:\Program Files\OpenToonz\imageformats\qwbmp.dll
C:\Program Files\OpenToonz\imageformats\qwebp.dll
C:\Program Files\OpenToonz\bearer\qgenericbearer.dll
C:\WINDOWS\SYSTEM32\dhcpcsvc6.DLL
C:\WINDOWS\SYSTEM32\dhcpcsvc.DLL
C:\WINDOWS\System32\netprofm.dll
C:\WINDOWS\System32\npmproxy.dll
C:\Program Files\OpenToonz\audio\qtaudio_windows.dll
C:\WINDOWS\SYSTEM32\MMDevAPI.DLL
C:\WINDOWS\SYSTEM32\wdmaud.drv
C:\WINDOWS\SYSTEM32\AVRT.dll
C:\WINDOWS\SYSTEM32\ksuser.dll
C:\WINDOWS\SYSTEM32\AUDIOSES.DLL
C:\WINDOWS\SYSTEM32\msacm32.drv
C:\WINDOWS\SYSTEM32\midimap.dll
C:\WINDOWS\SYSTEM32\resourcepolicyclient.dll
C:\WINDOWS\System32\mswsock.dll
C:\Windows\System32\rasadhlp.dll
C:\WINDOWS\system32\napinsp.dll
C:\WINDOWS\system32\pnrpnsp.dll
C:\WINDOWS\System32\winrnr.dll
C:\WINDOWS\system32\wshbth.dll
C:\WINDOWS\system32\nlansp_c.dll
C:\WINDOWS\System32\fwpuclnt.dll
C:\Windows\System32\Windows.System.Launcher.dll
C:\Windows\System32\msvcp110_win.dll
C:\WINDOWS\SYSTEM32\windows.staterepositorycore.dll
C:\WINDOWS\SYSTEM32\LINKINFO.dll
C:\Windows\System32\Windows.StateRepositoryPS.dll
C:\WINDOWS\SYSTEM32\windows.staterepositoryclient.dll
C:\Windows\System32\Windows.FileExplorer.Common.dll
C:\WINDOWS\system32\winhttp.dll
C:\WINDOWS\SYSTEM32\SspiCli.dll
==== End ====

==== Backtrace ====
  1> RtlGetLengthWithoutLastFullDosOrNtPathElement [0x7ff8ce1c7f5a] <ntdll.dll>
  2> RtlFindCharInUnicodeString [0x7ff8ce16e642] <ntdll.dll>
  3> KiUserExceptionDispatcher [0x7ff8ce1f403e] <ntdll.dll>
  4> StyleEditorGUI::ColorSlider::setColor [0x7ff89315e237] <toonzqt.dll>
  5> StyleEditor::setStyle [0x7ff89315f1aa] <toonzqt.dll>
  6> StyleEditor::onStyleSwitched [0x7ff89315b8fe] <toonzqt.dll>
  7> QObject::qt_static_metacall [0x7ff8914820d7] <Qt5Core.dll>
  8> TPaletteHandle::setPalette [0x7ff890e5663b] <toonzlib.dll>
  9> PaletteController::editLevelPalette [0x7ff890e76630] <toonzlib.dll>
 10> TSmartPointerT<TXshPaletteLevel>::~TSmartPointerT<TXshPaletteLevel> [0x7ff7e66cfb70] <OpenToonz.exe>
 11> QObject::qt_static_metacall [0x7ff8914820d7] <Qt5Core.dll>
 12> TSceneHandle::setScene [0x7ff890e56916] <toonzlib.dll>
 13> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff7e6892c28] <OpenToonz.exe>
 14> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff7e683e317] <OpenToonz.exe>
 15> QObject::qt_static_metacall [0x7ff8914820d7] <Qt5Core.dll>
 16> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff7e6a278c0] <OpenToonz.exe>
 17> QObject::qt_static_metacall [0x7ff8914820d7] <Qt5Core.dll>
 18> QListWidget::qt_static_metacall [0x7ff893cdaa1b] <Qt5Widgets.dll>
 19> QObject::qt_static_metacall [0x7ff8914820d7] <Qt5Core.dll>
 20> QAbstractItemView::mouseReleaseEvent [0x7ff893ca2a0f] <Qt5Widgets.dll>
 21> QListView::mouseReleaseEvent [0x7ff893cd346f] <Qt5Widgets.dll>
 22> QWidget::event [0x7ff893aa8092] <Qt5Widgets.dll>
 23> QFrame::event [0x7ff893b49d10] <Qt5Widgets.dll>
 24> QAbstractItemView::viewportEvent [0x7ff893ca8ae8] <Qt5Widgets.dll>
 25> QCoreApplicationPrivate::sendThroughObjectEventFilters [0x7ff89146523a] <Qt5Core.dll>
 26> QApplicationPrivate::notify_helper [0x7ff893a855ac] <Qt5Widgets.dll>
 27> QApplication::notify [0x7ff893a834e0] <Qt5Widgets.dll>
 28> QCoreApplication::notifyInternal2 [0x7ff89146318a] <Qt5Core.dll>
 29> QApplicationPrivate::sendMouseEvent [0x7ff893a867c5] <Qt5Widgets.dll>
 30> QDesktopWidget::workAreaResized [0x7ff893ad09d2] <Qt5Widgets.dll>
 31> QDesktopWidget::workAreaResized [0x7ff893ace96e] <Qt5Widgets.dll>
 32> QApplicationPrivate::notify_helper [0x7ff893a855c0] <Qt5Widgets.dll>
 33> QApplication::notify [0x7ff893a84643] <Qt5Widgets.dll>
 34> QCoreApplication::notifyInternal2 [0x7ff89146318a] <Qt5Core.dll>
 35> QGuiApplicationPrivate::processMouseEvent [0x7ff8918a47c3] <Qt5Gui.dll>
 36> QWindowSystemInterface::sendWindowSystemEvents [0x7ff89188f030] <Qt5Gui.dll>
 37> QEventDispatcherWin32::processEvents [0x7ff8914abcfa] <Qt5Core.dll>
 38> qt_plugin_query_metadata [0x7ff8939628d9] <qwindows.dll>
 39> QEventLoop::exec [0x7ff89145f64c] <Qt5Core.dll>
 40> QCoreApplication::exec [0x7ff891462164] <Qt5Core.dll>
 41> TSmartPointerT<TXshSoundLevel>::~TSmartPointerT<TXshSoundLevel> [0x7ff7e6685219] <OpenToonz.exe>
 42> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff7e6aaeab7] <OpenToonz.exe>
 43> TSmartPointerT<TXshSoundTextLevel>::~TSmartPointerT<TXshSoundTextLevel> [0x7ff7e6aae092] <OpenToonz.exe>
 44> BaseThreadInitThunk [0x7ff8ccf5257d] <KERNEL32.DLL>
 45> RtlUserThreadStart [0x7ff8ce1aaf28] <ntdll.dll>
==== End ====

### Steps to Reproduce

1. I opened opentoonz and started doing my animation
2. Something happen and I clicked on cleanup settings
3. Appeared Opentoonz crashed
4. Closed and I open it again
5. Appeared what happen
6. Now I'm here writing this

### OpenToonz Version

1.7

### Operating System

Windows

### GPU

NVIDIA

### Graphics Tablet

None

**Comment by RodneyBaker** [comment]
Crash report analyzis:

```The crash report indicates that OpenToonz experienced an EXCEPTION_ACCESS_VIOLATION, which typically occurs when the application attempts to access memory that it is not allowed to, such as accessing null or corrupted pointers. Here's a breakdown of the crash report and possible steps to diagnose and resolve the issue:

Key Points from the Crash Report
Crash Reason: The crash is due to EXCEPTION_ACCESS_VIOLATION. This error usually arises from accessing memory locations that the program doesn't own, often due to programming bugs such as dereferencing null or uninitialized pointers.

Backtrace Analysis:

The stack trace shows a series of function calls leading up to the crash, starting with RtlGetLengthWithoutLastFullDosOrNtPathElement in ntdll.dll.
The critical point appears to be in StyleEditorGUI::ColorSlider::setColor within toonzqt.dll, followed by functions like StyleEditor::setStyle and StyleEditor::onStyleSwitched.
The involvement of StyleEditor functions suggests the crash might be related to how styles or colors are being applied or managed.
Modules Loaded: A wide array of modules, including Qt libraries (Qt5Core, Qt5Widgets, etc.), OpenGL, and OpenToonz-specific DLLs (toonzqt.dll, toonzlib.dll), are loaded. The crash within toonzqt.dll might indicate an issue specific to the GUI handling within OpenToonz.

Steps to Reproduce:

You mentioned that the crash occurred when clicking on "cleanup settings" after starting animation work. This action likely triggered the bug related to the StyleEditor component.
Possible Causes and Solutions
Software Bugs:

Cause: The crash likely results from a bug in OpenToonz when handling style or color changes, possibly due to incorrect pointer management or uninitialized variables.
Solution: Check for updates or patches for OpenToonz, as this might be a known issue that has been fixed in newer versions.
Graphics Driver Issues:

Cause: Sometimes, crashes related to rendering or style editing can be caused by outdated or incompatible graphics drivers.
Solution: Ensure your GPU drivers are up-to-date. Consider rolling back to a previous stable driver version if the issue started after a recent driver update.
Corrupted Preferences or Settings:

Cause: Corruption in OpenToonz's configuration or style files might lead to crashes when accessing those settings.
Solution: Try resetting OpenToonz's preferences. Backup your settings and preferences first, then delete or rename the settings directory (e.g., C:\Users\<Username>\AppData\Local\OpenToonz) and restart the program.
Project or Scene File Issues:

Cause: There might be an issue with the specific project or scene file that triggers the crash.
Solution: Open a new project or a different scene to see if the crash persists. If it doesn't, the issue might be with the specific project file. Try recreating the project or importing elements into a new scene.
Memory and Resource Limitations:

Cause: Running out of memory or system resources can lead to access violations.
Solution: Close other applications to free up resources and ensure your system meets the recommended specifications for running OpenToonz.
Conclusion
Given the specific error location and the steps leading up to the crash, focusing on style editing and GUI-related functionality in OpenToonz is key. Checking for software updates, ensuring your graphics drivers are current, and verifying the integrity of your configuration and project files are practical first steps in resolving this issue. If the problem persists, consider reaching out to the OpenToonz community or support channels for further assistance.```

TLDR:  Consider updating to the nightly release (but also see:  #5351)

**Comment by RodneyBaker** [comment]
Closing as likely resolved and/or abandoned.
